### PR TITLE
[portal] Implement gas airdrop and add testsuit to gas market

### PR DESCRIPTION
**Pull request by jolestar**
## Summary

Implement gas airdrop and add testsuit to gas market

### CODE REVIEW

File: infra/rooch-portal-v2/contract/gas_market/sources/gas_airdrop.move [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+      let remaining_rgas_amount = coin_store::balance(&airdrop.rgas_store);
+      assert!(claim_rgas_amount <= remaining_rgas_amount, ErrorAirdropNotEnoughRGas);
+      let rgas_coin = coin_store::withdraw(&mut airdrop.rgas_store, claim_rgas_amount);
+      account_coin_store::deposit<RGas>(claimer, rgas_coin);
+    }
+
+    public entry fun deposit_rgas_coin(
+        account: &signer,
+        rgas_airdrop_obj: &mut Object<RGasAirdrop>,
```
Comment (mx819812523) [inline infra/rooch-portal-v2/contract/gas_market/sources/gas_airdrop.move:70]: 是否需要通过`object::named_object_id<RGasAirdrop>()`获取，前端少记录一个参数？

Comment (jolestar) [reply infra/rooch-portal-v2/contract/gas_market/sources/gas_airdrop.move:76]: 这个应该前端不用调用，我们需要充值的时候调用一下就行。另外 `claim` 方法我也加上 Object 参数了。少算一次 name 可以省一些 gas。

File: infra/rooch-portal-v2/contract/gas_market/sources/gas_airdrop.move [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      let rgas_airdrop_obj = object::new_named_object(RGasAirdrop{
+          rgas_store,
+          claim_records: table::new(),
+          is_open: true
+      });
+      to_shared(rgas_airdrop_obj)
+    }
+
+    /// Anyone can call this function to help the claimer claim the airdrop
+    public entry fun claim(airdrop_obj: &mut Object<RGasAirdrop>, claimer: address, utxo_ids: vector<ObjectID>){
```
Comment (baichuan3) [inline infra/rooch-portal-v2/contract/gas_market/sources/gas_airdrop.move:60]: Where is the height snapshot determined?

Comment (jolestar) [reply infra/rooch-portal-v2/contract/gas_market/sources/gas_airdrop.move:60]: We do not check the snapshot; any address that has UTXO can claim the gas coin.

**Review by baichuan3 — APPROVED** [review]

**Pull request merged**